shared/js/background/chrome-events.es6.js

```diff
@@ -106,7 +106,14 @@ chrome.webRequest.onHeadersReceived.addListener(
             const tab = tabManager.get({ tabId: request.tabId })
             if (!request.responseHeaders) return
             if (tab && tab.site.whitelisted) return
-            if (tab && utils.isFirstParty(request.url, tab.url)) return
+
+            if (!tab && request.tabId === -1) {
+                if (utils.isFirstParty(request.initiator, request.url)) {
+                    return
+                }
+            } else if (tab && utils.isFirstParty(request.url, tab.url)) {
+                return
+            }
```

Comment:
So if we don't have info on the tab (`!tab`) then we will always block tracker cookies. Will this not break when restoring a session, and the tabManager may not yet have tab info?

Reply:
We'd only be removing the cookie header for tracker urls which I believe is the policy here.

Reply:
Aren't some internal requests at least in Firefox in this category also?

Reply:
What I meant is that if `!tab && request.tabId > -1` and the url is a tracker URL we will still remove the cookie, without doing a first party check. I think we should handle that, just in case that happens and we end up blocking all the first party cookies on a site until the tab is updated.

Reply:
Fixed these scenarios. I encourage ya'll to give it a test 👍